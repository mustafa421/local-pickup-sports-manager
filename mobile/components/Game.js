//import libraries

import React from "react";
import { Text, View, SectionList, Dimensions, ScrollView } from "react-native";
import PropTypes from "prop-types";
import { Card, Button, Rating } from "react-native-elements";

//Create Game Component
function Game(props) {
  return (
    <View
      style={{
        backgroundColor: "lightblue",
        width: Dimensions.get("window").width,
        height: Dimensions.get("window").height
      }}
    >
      <Card
        title={props.title}
        titleStyle={{ fontSize: 32, fontWeight: "bold", color: "black" }}
        containerStyle={{
          backgroundColor: "#03A9F4",
          borderColor: "black"
        }}
      >
        <Text
          style={{
            marginBottom: 0,
            marginRight: 0,
            marginTop: 0,
            marginLeft: 0,
            fontWeight: "bold",
            fontSize: 32
          }}
        >
          {" "}
          {props.date + ", " + props.time}
        </Text>

        <Text
          style={{
            marginBottom: 0,
            marginRight: 0,
            marginTop: 0,
            marginLeft: 0,
            fontWeight: "bold",
            fontSize: 24
          }}
        >
          {"@" + props.location}
        </Text>
      </Card>
      <View style={{ flexDirection: "row" }}>
        <Text
          style={{
            paddingLeft: 15,
            fontWeight: "bold",
            fontSize: 30,
            paddingVertical: 5
          }}
        >
          Difficulty:
        </Text>
        <Rating
          type="star"
          fractions={1}
          ratingCount={10}
          startingValue={props.difficulty}
          readonly
          imageSize={15}
          onFinishRating={this.ratingCompleted}
          onStartRating={this.ratingStarted}
          style={{ paddingLeft: 15, paddingVertical: 20, paddingRight: 15 }}
        />
      </View>
      <SectionList
        containerStyle={{
          marginLeft: 10,
          marginTop: 10,
          marginBottom: 0,
          paddingVertical: 0
        }}
        renderItem={({ item, index, section }) => (
          <Text key={index} style={{ paddingLeft: 15 }}>
            {item}
          </Text>
        )}
        renderSectionHeader={({ section: { title } }) => (
          <Text style={{ fontWeight: "bold", fontSize: 30, paddingLeft: 15 }}>
            {title}
          </Text>
        )}
        sections={[
          { title: "Participants: ", data: [props.number_going] },
          { title: "Interested: ", data: [props.number_interested] } //user profiles could go here
        ]}
        keyExtractor={(item, index) => item + index}
      />
      <Button
        backgroundColor="paleturquoise"
        color="black"
        buttonStyle={{
          borderRadius: 5,
          marginLeft: 0,
          marginRight: 0,
          marginBottom: 0,
          borderColor: "black",
          borderWidth: 1
        }}
        title="Join Game"
        onPress={() =>
          joinGame({
            userID,
            gameID,
            name,
            interested: false
          })
        }
      />
      <Text />
      <Button
        backgroundColor="paleturquoise"
        color="black"
        buttonStyle={{
          borderRadius: 5,
          marginLeft: 0,
          marginRight: 0,
          marginBottom: 0,
          borderColor: "black",
          borderWidth: 1
        }}
        title="Interested in Game"
        onPress={() =>
          joinGame({
            userID,
            gameID,
            name,
            interested: true
          })
        }
      />
    </View>
  );
}

Game.propTypes = {
  title: PropTypes.string.isRequired,
  location: PropTypes.string.isRequired,
  date: PropTypes.string.isRequired,
  time: PropTypes.string.isRequired,
  number_going: PropTypes.number.isRequired,
  number_interested: PropTypes.number.isRequired,
  difficulty: PropTypes.number.isRequired
};
//Render the Game Component to the Device
export default Game;
