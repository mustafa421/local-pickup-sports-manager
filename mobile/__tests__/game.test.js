import React from 'react';
import Game from "../components/game";
import renderer from 'react-test-renderer';

it('renders correctly', () => {
    const tree = renderer
    .create(<Game
        location={"madison"}
        date={"Tomorrow"}
        time={"12:00"}
        number_interested={3}
        number_going={4}
        difficulty={1}/>)
        .toJSON();
        expect(tree).toMatchSnapshot();
}) ;

it('renders correctly again', () => {
    const tree = renderer
    .create(<Game
        location={"Courts"}
        date={"11/27"}
        time={"3 PM"}
        number_interested={1}
        number_going={9}
        difficulty={4}/>)
        .toJSON();
        expect(tree).toMatchSnapshot();
}) ;

//These should throw errors
it('rejects location', () => {
    const tree = renderer
    .create(<Game
        location={4}
        date={"11/27"}
        time={"3 PM"}
        number_interested={1}
        number_going={9}
        difficulty={4}/>)
        .toJSON();

        
        expect(tree).toMatchSnapshot();
          
}) ;

it('rejects date', () => {
    const tree = renderer
    .create(<Game
        location={"Madison"}
        date={4}
        time={"3 PM"}
        number_interested={1}
        number_going={9}
        difficulty={4}/>)
        .toJSON();

        
        expect(tree).toMatchSnapshot();
          
}) ;

it('rejects time', () => {
    const tree = renderer
    .create(<Game
        location={"Madison"}
        date={"November 1"}
        time={6}
        number_interested={1}
        number_going={9}
        difficulty={4}/>)
        .toJSON();

        
        expect(tree).toMatchSnapshot();
          
}) ;

it('rejects number interested', () => {
    const tree = renderer
    .create(<Game
        location={"Courts"}
        date={"11/27"}
        time={"3 PM"}
        number_interested={"1"}
        number_going={9}
        difficulty={4}/>)
        .toJSON();
        expect(tree).toMatchSnapshot();
}) ;

it('rejects number going', () => {
    const tree = renderer
    .create(<Game
        location={"Courts"}
        date={"11/27"}
        time={"3 PM"}
        number_interested={1}
        number_going={"9"}
        difficulty={4}/>)
        .toJSON();
        expect(tree).toMatchSnapshot();
}) ;

it('rejects difficulty', () => {
    const tree = renderer
    .create(<Game
        location={"Courts"}
        date={"11/27"}
        time={"3 PM"}
        number_interested={1}
        number_going={9}
        difficulty={"4"}/>)
        .toJSON();
        expect(tree).toMatchSnapshot();
}) ;