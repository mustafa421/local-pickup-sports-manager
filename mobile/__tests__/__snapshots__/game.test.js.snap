// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rejects date 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
    }
  }
>
  <Text>
    Location: 
    Madison
     
  </Text>
  <Text>
    Date: 
    4
     
  </Text>
  <Text>
    Time: 
    3 PM
     
  </Text>
  <Text>
    Going: 
    9
     
  </Text>
  <Text>
    Interested: 
    1
     
  </Text>
  <Text>
    Difficulty: 
    4
     
  </Text>
</View>
`;

exports[`rejects difficulty 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
    }
  }
>
  <Text>
    Location: 
    Courts
     
  </Text>
  <Text>
    Date: 
    11/27
     
  </Text>
  <Text>
    Time: 
    3 PM
     
  </Text>
  <Text>
    Going: 
    9
     
  </Text>
  <Text>
    Interested: 
    1
     
  </Text>
  <Text>
    Difficulty: 
    4
     
  </Text>
</View>
`;

exports[`rejects location 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
    }
  }
>
  <Text>
    Location: 
    4
     
  </Text>
  <Text>
    Date: 
    11/27
     
  </Text>
  <Text>
    Time: 
    3 PM
     
  </Text>
  <Text>
    Going: 
    9
     
  </Text>
  <Text>
    Interested: 
    1
     
  </Text>
  <Text>
    Difficulty: 
    4
     
  </Text>
</View>
`;

exports[`rejects number going 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
    }
  }
>
  <Text>
    Location: 
    Courts
     
  </Text>
  <Text>
    Date: 
    11/27
     
  </Text>
  <Text>
    Time: 
    3 PM
     
  </Text>
  <Text>
    Going: 
    9
     
  </Text>
  <Text>
    Interested: 
    1
     
  </Text>
  <Text>
    Difficulty: 
    4
     
  </Text>
</View>
`;

exports[`rejects number interested 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
    }
  }
>
  <Text>
    Location: 
    Courts
     
  </Text>
  <Text>
    Date: 
    11/27
     
  </Text>
  <Text>
    Time: 
    3 PM
     
  </Text>
  <Text>
    Going: 
    9
     
  </Text>
  <Text>
    Interested: 
    1
     
  </Text>
  <Text>
    Difficulty: 
    4
     
  </Text>
</View>
`;

exports[`rejects time 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
    }
  }
>
  <Text>
    Location: 
    Madison
     
  </Text>
  <Text>
    Date: 
    November 1
     
  </Text>
  <Text>
    Time: 
    6
     
  </Text>
  <Text>
    Going: 
    9
     
  </Text>
  <Text>
    Interested: 
    1
     
  </Text>
  <Text>
    Difficulty: 
    4
     
  </Text>
</View>
`;

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
    }
  }
>
  <Text>
    Location: 
    madison
     
  </Text>
  <Text>
    Date: 
    Tomorrow
     
  </Text>
  <Text>
    Time: 
    12:00
     
  </Text>
  <Text>
    Going: 
    4
     
  </Text>
  <Text>
    Interested: 
    3
     
  </Text>
  <Text>
    Difficulty: 
    1
     
  </Text>
</View>
`;

exports[`renders correctly again 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
    }
  }
>
  <Text>
    Location: 
    Courts
     
  </Text>
  <Text>
    Date: 
    11/27
     
  </Text>
  <Text>
    Time: 
    3 PM
     
  </Text>
  <Text>
    Going: 
    9
     
  </Text>
  <Text>
    Interested: 
    1
     
  </Text>
  <Text>
    Difficulty: 
    4
     
  </Text>
</View>
`;
